### DISCUSSION TITLE
How to initialise single object from map and key-value pair?

### DISCUSSION THREAD

**Discussion by fixgoats** [nlohmann/json#3611] [Q&A] [answered]
What I would like to do is something like
```
std::map<std::string, int> amap{{"a key", 0}};
json j{amap, {"some other key", 1}};
```
and have the output be `[{"a key": 0, "some other key": 1}]`. But when I try that, I get `[{"a key": 0}, ["some other key", 1]]`. Now, I can initialise `j` with just the map first, and then do `j["some other key"] = 1` and get the desired output, but I think it would be great to be able to do this at initialisation.

**Comment by falbrechtskirchinger** [comment, marked as answer]
I don't think that's currently possible but shouldn't be too difficult to add.
However, you can do the following:
```cpp
    std::map<std::string, int> amap{{"a key", 0}};
    json j{{"some other key", 1}};
    j.update(json(amap));
```

(Note that `json(amap)` and `json{amap}` produce different results!)

See https://json.nlohmann.me/api/basic_json/update/ for details.

**Reply by fixgoats** [reply]
Great, thanks for the quick answer and helpful suggestion.
